docs/DeveloperGuide.md

```diff
@@ -1,22 +1,118 @@
-# Developer Guide
+# AeroCade Developer Guide
 
-## Acknowledgements
+## Table of Contents
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [Design & Implementation](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#design--implementation)
+- [Product Scope](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#product-scope)
+- [User Stories](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#user-stories)
+- [Non-Functional Requirements](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#non-functional-requirements)
+- [Glossary](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#glossary)
+- [Instructions for Manual Testing](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#instructions-for-manual-testing)
 
-## Design & implementation
+## Acknowledgements
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Design & Implementation
+
+Here is a quick overview of the main components and how they interact with each other:
+
+- [Parser](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#parser-implementation): Reads input
+- [Ui](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#ui-implementation): The UI of the App
+- [Game](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#game-implementation): The Interface concerning the games
+- [TicTacToe](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#tictactoe-implementation): Game 1
+- [Hangman](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#hangman-implementation): Game 2
+
+### <span style="color:grey;">Parser Implementation:</span>
+Parser Class (v1.0) reads user input from the standard input stream, checks the input, 
+tests the validity, and conveys the validity back to the class it is called from.
+![img.png](img.png)
```

Comment:
Could change UML diagram for Parser class to have standard formatting e.g., remove java & io boxes around BufferedReader class.